### DISCUSSION TITLE
Pre-trained yolo v5-v8 model on VisDrone dataset?

### DISCUSSION THREAD

**Discussion by hellonun** [ultralytics/ultralytics#1255] [General]
Is there a pre-trained yolov5 or yolov8 model on VisDrone dataset I can find anywhere?

**Comment by AyushExel** [comment]
Currently, we only have coco pre-trained models But we might offer more pre-trained models in the future

**Comment by glenn-jocher** [comment]
@hellonun you can train VisDrone easily yourself:

```
yolo train model=yolov8n.pt data=VisDrone.yaml
```

**Reply by abhinav70291** [reply]
can anyone suggest a good hyperparamter configuration values  because my yolov8 model when trained for 25 epochs  on visdrone, still mAP 0.5 0.95 saturates at  around 0.05.

**Reply by glenn-jocher** [reply]
@abhinav70291 25 seems low, you want to train until your val losses begin to overfit.

**Reply by ai-1976** [reply]
how do I know when to stop training?

**Reply by abhinav70291** [reply]
I think the best way is to stop training when the mAP score reaches saturation.I think that u can set a threshold for early stopping rounds to around 50 epochs and take the best configuration from it.

Also, u might want to change some input hyperparameters and try to train the model again and see if the score gets any better.

Personally, I haven't found a solid criteria to know when to stop.For the visdrone dataset,it took me a little more than 150 epochs (probably less than 200) with image size set to 720 and augmentations being turned on.Try using the cosine scheduler and increasing warmup epochs.

**Reply by darwintk** [reply]
@abhinav70291 hi~ my visdrone dataset model also got low mAP50 and mAP50-95, would  you mind sharing your yolov8 hyper parameters?

**Reply by 1nyourlife** [reply]
Hi, sorry to bother but my found this error in my output:
FileNotFoundError: train: No labels found in /home/zl481/PycharmProjects/pythonProject1/datasets/VisDrone/VisDrone2019-DET-train/labels.cache. 
I have follow the tutorial and think the location of the dataset is correct, but still can not training,

**Reply by pderrenger** [reply]
@1nyourlife it appears that your dataset labels are not correctly located or formatted. Ensure that your labels are in the expected directory and that they are in the correct format. If you're using a custom dataset, you should have a `labels` folder alongside your `images` folder, with one text file for each image, containing the annotations.

If you're certain the labels are correctly placed, try clearing any cache files like `labels.cache` and re-running the training command. The dataset will be re-scanned, and a new cache will be created. If the issue persists, double-check your dataset configuration file (YAML) to ensure the paths are correctly specified.

**Comment by yrik** [comment]
You can find pre-trained VisDrone models here https://huggingface.co/mshamrai/yolov8n-visdrone

**Reply by pderrenger** [reply]
@yrik i'm glad to see the community sharing models trained on various datasets! However I should clarify that the official pre-trained models we provide are typically trained on widely-used datasets such as COCO. For specialized datasets like VisDrone, we rely on the community to train and share their models.

If you're interested in training a YOLOv8 model on the VisDrone dataset yourself, you can follow our training guide in the Ultralytics Docs. This will involve setting up your dataset in the correct format, adjusting the training configuration, and then using the training command to train your model. If you need any assistance during the process, feel free to raise an issue in the repo, and we'll be happy to help! 😊🚀

**Comment by agoncharenko1992** [comment]
Hi there! 

Here you can find a little bit better solutions for VisDrone dataset:
 https://huggingface.co/ENOT-AutoDL/yolov8s_visdrone
 (+8% against Ultralitics baseline cfg.)

**Reply by pderrenger** [reply]
@agoncharenko1992 thank you for your interest in the YOLO models! 🚀

Currently, we do not provide pre-trained YOLOv8 models on the VisDrone dataset within our official Ultralytics repositories. However, you can train a YOLOv8 model on the VisDrone dataset yourself by following the training instructions in our documentation. You would need to prepare the VisDrone dataset in the required format and then use the training command with the appropriate dataset configuration.

For detailed guidance on how to train a YOLOv8 model on a custom dataset, please refer to our training documentation at [Train](https://docs.ultralytics.com/modes/train). If you have any further questions or need assistance, feel free to raise an issue in the repo, and we'll be happy to help! Happy coding! 🤖💻
